tomviz/python/PeronaMalikAnisotropicDiffusion.py

```diff
@@ -32,8 +32,23 @@ def transform_scalars(self, dataset, conductance=1.0, iterations=100,
             # Get the ITK image. The itk.GradientAnisotropicDiffusionImageFilter
             # is templated over float pixel types only, so explicitly request a
             # float ITK image type.
-            itk_image = itkutils.convert_vtk_to_itk_image(dataset, itkTypes.F)
-            itk_image_type = type(itk_image)
+            itk_image = itkutils.convert_vtk_to_itk_image(dataset)
```

Comment:
This will create a new VTK data object if the input `vtkImageData` has no direct ITK image type counterpart, so this doesn't completely solve the problem you report.